### PR TITLE
Refactor/favorite routes

### PR DESCRIPTION
**Pull request by TylerRockwell**
### Why?

https://github.com/smashingboxes/apprenticeship/tree/master/Curriculum/Backend/twitter
Because RESTful routes make things easier for everyone involved
### What Changed?

A FavoritesController was added with create/destroy actions
The user show path was changed from profile/username to user/username

### CODE REVIEW

**Comment by jahammo2** [comment]
Make sure you rerun your test suite right before submitting a PR. We actually have a tool we use on our big projects called Travis that will do this for us so you don't have to manually do it every time. But you do have a failing test right now.

```
  1) Favorite.destroy_favorite(user_id, post_id) should destroy a user favorite
     Failure/Error:
       expect { Favorite.destroy_favorite(current_user.id, post.id) }.
         to change(Favorite, :count).by(-1)

     NoMethodError:
       undefined method `destroy_favorite' for #<Class:0x007fe219ea6e30>
     # ./spec/models/favorite_spec.rb:18:in `block (4 levels) in <top (required)>'
     # ./spec/models/favorite_spec.rb:18:in `block (3 levels) in <top (required)>'
```

File: app/controllers/favorites_controller.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+class FavoritesController < ApplicationController
+  before_filter :authenticate_user!
+  before_action :set_post
+
+  def create
+    favorite = Favorite.new(user_id: current_user.id, post_id: @post.id)
```
Comment (jahammo2) [inline app/controllers/favorites_controller.rb]: You don't actually need to call the instance variable of `@post`. You can just call `post`.

File: app/controllers/favorites_controller.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    flash[:alert] = "Could not save favorite" unless favorite.save
+    redirect_to posts_path
+  end
+
+  def destroy
+    favorite = Favorite.find_favorite(current_user.id, @post.id)
```
Comment (jahammo2) [inline app/controllers/favorites_controller.rb]: Same

**Comment by jahammo2** [comment]
Get that test passing so you can :shipit:

File: app/controllers/favorites_controller.rb (hunk 1 above)
Comment (TylerRockwell) [inline app/controllers/favorites_controller.rb]: I just took the post object out. This controller only ever uses the id of post, so I don't think there's a need to even look up post in the first place. Thanks for pointing this out.

**Comment by TylerRockwell** [comment]
Good call on the test suite. I forgot to run it right before the last commit. It's all good now. Annie says I will start using Travis here soon. Thanks for the review!

**Pull request merged**